src/pip/__main__.py

```diff
@@ -21,4 +21,12 @@
 from pip._internal.cli.main import main as _main
 
 if __name__ == "__main__":
+    try:
+        # windows multiprocessing support
+        from multiprocessing import freeze_support
+
+        freeze_support()
+    except ImportError:
+        pass
```

Comment:
```suggestion
    # windows single-executable multiprocessing support
    from multiprocessing import freeze_support

    freeze_support()
```

According to the documentation, `freeze_support()` is always available, and is simply a no-op on non-Windows platforms.